Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -341,6 +351,7 @@ open class NavigationMapView: UIView {
         mapView = MapView(frame: frame, mapInitOptions: mapInitOptions)
         mapView.translatesAutoresizingMaskIntoConstraints = false
         mapView.ornaments.options.scaleBar.visibility = .hidden
+        storeLocationProvider()
```

Comment:
Does it affect the free-driving? It's when you override the location provider with `PassiveLocationProvider` after a `NavigationMapView` is created.

Reply:
Right now the `locationProvider` is only responsible to store the `locationProvider`. The only case that it would be used, is that when the `simulatesLocation` sets to `false` and it happens after the active navigation starts or the simulation mode of navigation service turned off. Because it's related with the `NavigationLocationManager` during active driving. 

During free drive, there's no change of the `simulatesLocation` of the `navigationMapView`. And if the user decides to use their own location provider for the `mapView` during active navigation, it works after tests.